### DISCUSSION TITLE
Export layer area to a file

### DISCUSSION THREAD

**Discussion by enggraph** [sn4k3/UVtools#192] [Scripts]
Hello,

it is possible to export the calculated layer area into a text file?
I need a file with all the layers (1 row / 1layer)

![export](https://user-images.githubusercontent.com/83723836/117187120-dc46c280-addb-11eb-916e-d213bb8456ac.JPG)

Thank You
Levi

**Comment by sn4k3** [comment]
Yes, you can do that easy via Scripting and add the data you want to the text file you want: https://github.com/sn4k3/UVtools/tree/master/UVtools.ScriptSample
https://docs.microsoft.com/en-us/dotnet/api/system.io.file.createtext?view=net-5.0

The property on UVtools to get that area in mm is: `layer.BoundingRectangleMillimeters.Area()`
So you just need to loop all layers and export the area per line

Best script sample to start with for what you want: https://github.com/sn4k3/UVtools/blob/master/UVtools.ScriptSample/ScriptChangeLayerPropertiesSample.cs

**Reply by enggraph** [reply]
Hello sn4k3

Thank You for your help.
i am an enduser from uv-tools and not a programmer :-( but i will try it

Levi

**Reply by sn4k3** [reply]
Well i could write you the script in no time, but take a shot on it by yourself. Once you start playing with it and able to do your own stuff you will reward from that and open doors to other scripts you may want to do.
If you need help just ask

**Reply by enggraph** [reply]
Hello Sn4k3,

I have been trying to create the script, but i have a problem with the layer.BoundingRectangleMillimeters.Area property. 

![UV_Failure](https://user-images.githubusercontent.com/83723836/117369772-b1389d80-aec5-11eb-95e0-c50748868e03.JPG)

 I used another properties like layer.NonZeroPixelCount, and its works but with the area i have some problems.
Can you help me?

my standat time:

```C#
/*
 *                     GNU AFFERO GENERAL PUBLIC LICENSE
 *                       Version 3, 19 November 2007
 *  Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
 *  Everyone is permitted to copy and distribute verbatim copies
 *  of this license document, but changing it is not allowed.
 */

using System;
using UVtools.Core.Scripting;
using System.IO;

namespace UVtools.ScriptSample
{
    /// <summary>
    /// Change layer properties to random values
    /// </summary>
    public class ScriptChangeLayerPropertiesSample : ScriptGlobals
    {
        /// <summary>
        /// Set configurations here, this function trigger just after load a script
        /// </summary>
        public void ScriptInit()
        {
            Script.Name = "Change layer properties";
            Script.Description = "Change layer properties to random values :D";
            Script.Author = "Tiago Conceição";
            Script.Version = new Version(0, 1);
        }

        /// <summary>
        /// Validate user inputs here, this function trigger when user click on execute
        /// </summary>
        /// <returns>A error message, empty or null if validation passes.</returns>
        public string ScriptValidate()
        {
            return null;
        }

        /// <summary>
        /// Execute the script, this function trigger when when user click on execute and validation passes
        /// </summary>
        /// <returns>True if executes successfully to the end, otherwise false.</returns>
        public bool ScriptExecute()
        {
            string path = @"c:\temp\UVToolAreaExp_" + DateTime.Now.ToString("yyyyMMddHHmmss") + ".txt";
           
            Progress.Reset("Changing layers", Operation.LayerRangeCount); // Sets the progress name and number of items to process
            
            Random random = new();
            using (StreamWriter sw = File.CreateText(path))
            for (uint layerIndex = Operation.LayerIndexStart; layerIndex <= Operation.LayerIndexEnd; layerIndex++)
            {
                Progress.Token.ThrowIfCancellationRequested(); // Abort operation, user requested cancellation
                var layer = SlicerFile[layerIndex]; // Unpack and expose layer variable for easier use
               
               
                
               sw.WriteLine(layerIndex  + @"\" + layer.NonZeroPixelCount + @"\" + layer.BoundingRectangleMillimeters.Area);
               //sw.WriteLine(SlicerFile.GetName);
            

                Progress++; // Increment progress bar by 1
                
            }

            // return true if not cancelled by user
            return !Progress.Token.IsCancellationRequested;
        }
    }
}

```

**Reply by sn4k3** [reply]
Yes, `.Area` is a function not a property... use `.Area()` instead

Also that .Area is part of an extension, you need to declare a using on top near the others: `using UVtools.Core.Extensions;`

As bonus, change your line by my improved line:

`sw.WriteLine($@"{layerIndex}\{layer.NonZeroPixelCount}\{layer.BoundingRectangleMillimeters.Area()}");`

After you write the script i can review it and optimize it for you

**Reply by enggraph** [reply]
Thank You !!!!
Now it work's

**Reply by sn4k3** [reply]
How does it feels when we accomplish stuff by ourselves ;)

**Reply by enggraph** [reply]
very cool ;-)
